unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import uuid
+import json
+from unittest.mock import MagicMock, patch
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+AREA_ID = str(uuid.uuid4())
+TRANSACTION_ID = str(uuid.uuid4())
+DEVICE_ID = str(uuid.uuid4())
+
+
+# pylint: disable=protected-access
+class TestRedisClientBase:
+    """Tests for the redis client base behaviour without connecting to d3a."""
+
+    @staticmethod
+    @pytest.fixture()
+    def redis_client_auto_register():
+        return RedisClientBase(area_id=AREA_ID, autoregister=False)
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe")
+    def test_subscribe_to_response_channels(pubsub_mock, redis_client_auto_register):
+        """Check whether the pubsub.psubcribe is called with correct arguments."""
+        redis_client_auto_register._subscribe_to_response_channels()
+        redis_client_auto_register.pubsub.psubscribe.assert_called_with(
+            **{f"{AREA_ID}/response/register_participant": redis_client_auto_register._on_register,
+               f"{AREA_ID}/response/unregister_participant":
+                   redis_client_auto_register._on_unregister,
+               f"{AREA_ID}/*": redis_client_auto_register._on_event_or_response,
+               "aggregator_response": redis_client_auto_register._aggregator_response_callback})
+
+    @staticmethod
+    def test_aggregator_response_callback(redis_client_auto_register):
+        """Check whether the transaction id buffer is popping out and returning the empty list."""
+        data = {"transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._transaction_id_buffer.append(TRANSACTION_ID)
+        redis_client_auto_register._aggregator_response_callback(message)
+        assert redis_client_auto_register._transaction_id_buffer == []
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id(redis_client_auto_register):
+        """Check the buffer message matching the command and id and returns as None."""
+        data = {"transaction_id": TRANSACTION_ID}
+        redis_client_auto_register._blocking_command_responses = (
+            {"register": {"transaction_id": TRANSACTION_ID}})
+        assert redis_client_auto_register._check_buffer_message_matching_command_and_id(
+            data) is None
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id_throws_exception(
+            redis_client_auto_register):
+        """Checks if _check_buffer_message_matching_command_and_id throws
+           exception if the message does not contain transaction_id."""
```

Comment:
Nit:
```suggestion
        """
        Checks if _check_buffer_message_matching_command_and_id throws exception if the message
        does not contain transaction_id.
        """
```

Reply:
I think you don't necessarily need to state the full name of the method in the docstring (if you find a better way to rephrase it). This might help with keeping the docstring shorter.

Reply:
Sure, I will rephrase the docstring to 99 characters rather than double lines.